app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -16,24 +16,30 @@ def find_destination_in_vmdb(vm_uid_hash)
 
   def gather_storage_account_properties
     sas = nil
+
     source.with_provider_connection do |azure|
       sas = Azure::Armrest::StorageAccountService.new(azure)
     end
+
     return if sas.nil?
 
     begin
-      key             = sas.list_account_keys(storage_account_name, storage_account_resource_group).fetch('key1')
-      storage_account = sas.get(storage_account_name, storage_account_resource_group)
-      blob            = storage_account.blobs("system", key).first
-      blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
-      endpoint        = storage_account.properties.primary_endpoints.blob
-      source_uri      = File.join(endpoint, blob.container, blob.name)
-      target_uri      = File.join(endpoint, "manageiq", dest_name + "_" + SecureRandom.uuid + ".vhd")
+      image = sas.list_private_images(storage_account_resource_group).find do |img|
+        img.storage_account.name == storage_account_name
+      end
```

Comment:
@djberg96 - So this will find the first of possibly many images returned by the `find` method, right?
I am now wondering if we can find the image in Azure using the `source.ems_ref`property, where `source` is the image the user is provisioning an instance from.
That would guarantee we get the correct image and this method would be simplified even more.